File: malevich-server/src/main/java/io/malevich/server/transfer/OrderDto.java
package io.malevich.server.transfer;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonView;
import com.yinyang.core.server.core.dto.View;
import lombok.Getter;
import lombok.Setter;


@Getter
@Setter
public class OrderDto {

    @JsonView({View.Admin.class, View.Gallery.class, View.Trader.class})
    private String id;

    private OrderTypeDto type;

    private ArtworkStockDto artworkStock;

    private ParticipantDto party;

    private TradeTypeDto tradeType;

    private Double amount;

    @JsonView({View.Admin.class})
    private java.sql.Timestamp effectiveDate;

    private Double bestBid;

    private Double currentAsk;

    @JsonFormat(pattern = "dd/MM/yyyy")
    private java.sql.Timestamp expirationDate;

}

File: malevich-web/src/app/_transfer/orderDto.ts
import {OrderTypeDto} from "./orderTypeDto";
import {ArtworkStockDto} from "./artworkStockDto";
import {TradeTypeDto} from "./tradeTypeDto";
import {ParticipantDto} from "./participantDto";

export class OrderDto {
  id: string;
  type: OrderTypeDto;
  artworkStock: ArtworkStockDto;
  party: ParticipantDto;
  tradeType: TradeTypeDto;
  amount: number;
  effectiveDate: Date;
  expirationDate: Date;
}
